x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -314,17 +315,23 @@ export class TaskStore {
    * @returns {Promise<TaskDoc>}
    */
   public async update(doc: ConcreteTaskInstance): Promise<ConcreteTaskInstance> {
-    const updatedSavedObject = await this.savedObjectsRepository.update(
-      'task',
-      doc.id,
-      taskInstanceToAttributes(doc),
-      {
-        refresh: false,
-        version: doc.version,
-      }
-    );
+    const attributes = taskInstanceToAttributes(doc);
+    const updatedSavedObject = await this.savedObjectsRepository.update<
+      SerializedConcreteTaskInstance
+    >('task', doc.id, attributes, {
+      refresh: false,
+      version: doc.version,
+    });
 
-    return savedObjectToConcreteTaskInstance(updatedSavedObject);
+    const c = savedObjectToConcreteTaskInstance(
+      // The SavedObjects update api forces a Partial on the `attributes` on the response,
+      // but actually returns the whole object that is passed to it, so as we know we're
+      // passing in the whole object, this is safe to do.
+      // This is far from ideal, but unless we change the SavedObjectsClient this is the best we can do
+      // (updatedSavedObject as unknown) as SavedObject<SerializedConcreteTaskInstance>
+      { ...updatedSavedObject, attributes: defaults(updatedSavedObject.attributes, attributes) }
+    );
+    return c;
```

Comment:
This is what we discussed on the call yesterday - the typing in SOs basically forces a `Partial<>` where it isn't needed.
In case SOs changes it's underlying implementation in the future, we use defaults to ensure we still have all the required fields after update is applied.